oshmem/mca/memheap/base/memheap_base_alloc.c

```diff
@@ -90,3 +99,17 @@ int mca_memheap_alloc_with_hint(size_t size, long hint, void** ptr)
 
     return MCA_MEMHEAP_CALL(alloc(size, ptr));
 }
+
+map_segment_t *mca_memheap_base_allocate_segment(mca_memheap_map_t *map)
+{
+    assert(map->n_segments <= map->capacity);
+
+    if (map->n_segments >= map->capacity) {
+        map->capacity = map->capacity ? (map->capacity * 2) :
+                                        MCA_MEMHEAP_MAX_SEGMENTS;
```

Comment:
1. shall we limit by mca_memheap_base_max_segments here?
2. seems need to use MCA_MEMHEAP_MIN_SEGMENTS (new define) instead of MCA_MEMHEAP_MAX_SEGMENTS, and increase MCA_MEMHEAP_MAX_SEGMENTS to much larger value. because of dynamic allocation, we should not limit the default number of segments, so it will work OOB.